src/Streamly/Data/Unfold.hs

```diff
@@ -63,9 +63,13 @@ module Streamly.Data.Unfold
     -- One to one correspondence with
     -- "Streamly.Internal.Data.Stream.IsStream.Generate"
 
+    -- ** Basic Constructors
+    , unfoldrM
+    , function
+    , functionM
```

Comment:
`singleton` also made sense. It also indicates the number of elements in the stream. I would prefer `singleton` over `function` but this is also fine I guess. Up to you.

Reply:
Ok, scratch that. `singleton` can be confused with `yield`. `function` sounds good.